### PR TITLE
Update lisk-codec to check int range during write

### LINKED ISSUE #8969: Update CrossChainTransferCommand of NFT module
### Description

- Update CrossChainTransferCommand verification and execution of NFT module per [LIP PR](https://github.com/LiskHQ/lips/pull/477) to handle the case when there is no fee charged if `ccm.status != CCM_STATUS_CODE_OK`
- Also add the receivingChainID and sendingChainID in the ccmTransferEventDataSchema for the event EVENT_NAME_CCM_TRANSFER event and update the corresponding code in execution of the cross-chain transfer CCM.

### Acceptance Criteria
- All the relevant changes from the LIP PR must be incorporated
- Relevant unit tests should be updated

### Additional Information
[LIP PR](https://github.com/LiskHQ/lips/pull/477)

### PR DESCRIPTION
**Pull request by Phanco**
### What was the problem?

This PR resolves #8969

### CODE REVIEW

File: framework/test/unit/modules/interoperability/mainchain/commands/terminate_sidechain_for_liveness.spec.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ describe('TerminateSidechainForLivenessCommand', () => {
 				lastCertificate: {
 					height: 10,
 					stateRoot: utils.getRandomBytes(32),
-					timestamp: Date.now(),
+					timestamp: Math.floor(Date.now() / 1000),
```
Comment (shuse2) [inline framework/test/unit/modules/interoperability/mainchain/commands/terminate_sidechain_for_liveness.spec.ts:71]: this is not related to this issue?

File: framework/test/unit/modules/token/init_genesis_state_fixture.ts [hunk 2]
```diff
@@ -135,22 +135,6 @@ export const invalidGenesisAssets = [
 		},
 		"tokenID' maxLength exceeded",
 	],
-	[
```
Comment (shuse2) [inline framework/test/unit/modules/token/init_genesis_state_fixture.ts:138]: same as above

**Review by shuse2 — CHANGES_REQUESTED** [review]
change looks good to me but it includes unrelated changes

File: framework/test/unit/modules/interoperability/mainchain/commands/terminate_sidechain_for_liveness.spec.ts (hunk 1 above)
Comment (Phanco) [reply framework/test/unit/modules/interoperability/mainchain/commands/terminate_sidechain_for_liveness.spec.ts:71]: timestamp here should be without milliseconds otherwise it will exceed 2**64-1.

Also this timestamp compares with header timestamp which also uses w/o ms format

File: framework/test/unit/modules/token/init_genesis_state_fixture.ts (hunk 2 above)
Comment (Phanco) [reply framework/test/unit/modules/token/init_genesis_state_fixture.ts:138]: This test checks `initGenesis`, which decode a Buffer into object, and expects a "uint64 overflow".

However, since it's impossible to encode a value that exceeds uint64 limit. I think it is safe to remove this check and let the unit test above to handle the test

Comment (shuse2) [reply framework/test/unit/modules/token/init_genesis_state_fixture.ts:138]: i see it's failing. technically it's possible to have more value than uint64 if someone uses different encoder, but i guess it's fine to leave it out

**Review by shuse2 — APPROVED** [review]

**Review by ishantiw — APPROVED** [review]
